examples/thread-duell/main.c

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2020 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <thread.h>
+#include <sched.h>
+
+#include <math.h>
+
+#include "xtimer.h"
+
+#include "sched_round_robin.h"
+
+#define PRINT_STEPS 10
+#define WORK_SCALE  1000
+
+void bad_wait(uint64_t us)
+{
+    /*keep the cpu busy waiting for some time to pass simulate working*/
+    uint64_t until = xtimer_now_usec64() + us;
+
+    while (xtimer_now_usec64() < until) {
+        /*do nothing*/
+    }
+}
+
+void nice_wait(uint64_t us)
+{
+    /*be nice give the cpu some time to do other things or rest*/
+    xtimer_usleep64(us);
+}
+
+/*
+ * the following are threads that count and wait with different strategies and
+ * print their current count in steps.
+ * the ration of active (doing hard work like checking the timer)
+ * to passive (wait to be informed when a certain time is there) waiting
+ * is determint by there value given to the thread.
+ * the restless threads do never pause.
+ */
+
+void * thread_nicebreaks(void * d)
+{
+    xtimer_usleep64(200 * 1000);  /*always be nice at start*/
+#ifdef DEVELHELP
+    const char *name = thread_get_active()->name;
+#else
+    int16_t pid = thread_getpid();
+#endif
+
+    uint32_t w = 0;
+    uint32_t work = *((uint32_t *) d);
+    if (work > 10) {
+        work = 5;
+    }
+    uint32_t rest = (10 - work);
+    uint32_t step = 0;
+
+    /*work some time and rest*/
+    for (;;) {
+        if (w - step >= PRINT_STEPS) {
+            printf(
```

Comment:
this is ugly enough, maybe just have two whole printf lines?

Reply:
The whole loop is copied multiple times in this file, no? maybe factor out?

Reply:
i don't think reducing the number of copies that would make this more readable, but i took the two printf advice